### PR TITLE
Clesaec/tdi 45370 common test

### PR DESCRIPTION
**Pull request by clesaec**
https://jira.talendforge.org/browse/TDI-45370

Add a common module that help to design unit test.
This for Record & Assertions generator.

### CODE REVIEW

File: common-stream-io/stream-json/src/test/java/org/talend/components/common/stream/output/json/JsonExpected.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            Assertions.assertEquals(expectedValue, value);
+        }
+        else if (expectedValue instanceof Long) {
+            final long value = obj.getJsonNumber(name).longValue();
+            Assertions.assertEquals(expectedValue, value);
+        }
+        else if (expectedValue instanceof Record) {
+            final JsonObject value = obj.getJsonObject(name);
+            Assertions.assertNotNull(value);
+        }
```
Comment (chmyga) [inline common-stream-io/stream-json/src/test/java/org/talend/components/common/stream/output/json/JsonExpected.java:69]: Check of some possible types is missing

File: common-stream-io/stream-avro/src/test/java/org/talend/components/common/stream/output/avro/AvroExpected.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        }
+        else if (field.getType() == Schema.Type.RECORD) {
+            Assertions.assertNotNull(value);
+            Assertions.assertTrue(value instanceof GenericRecord);
+        }
+        else if (field.getType() != Schema.Type.DATETIME
+                && field.getType() != Schema.Type.BYTES) {
+            Assertions.assertNotNull(value);
+            Assertions.assertEquals(value.getClass(), expectedValue.getClass());
+        }
```
Comment (chmyga) [inline common-stream-io/stream-avro/src/test/java/org/talend/components/common/stream/output/avro/AvroExpected.java:75]: Check of some possible types are missing

**Review by chmyga — CHANGES_REQUESTED** [review]
If we add those classes we need to check all possible data types in them. Maybe it would be easier to drop them and check everything manually?

File: common-stream-io/stream-json/src/test/java/org/talend/components/common/stream/output/json/JsonExpected.java (hunk 1 above)
Comment (clesaec) [reply common-stream-io/stream-json/src/test/java/org/talend/components/common/stream/output/json/JsonExpected.java:98]: tests added

File: common-stream-io/stream-avro/src/test/java/org/talend/components/common/stream/output/avro/AvroExpected.java (hunk 2 above)
Comment (clesaec) [reply common-stream-io/stream-avro/src/test/java/org/talend/components/common/stream/output/avro/AvroExpected.java:90]: tests added

**Review by chmyga — APPROVED** [review]
LGTM

**Pull request merged**